--- ./src/xercesc/util/TransService.cpp.orig	2018-02-15 02:22:38.000000000 +0100
+++ ./src/xercesc/util/TransService.cpp	2018-06-09 00:49:47.000000000 +0200
@@ -42,6 +42,10 @@
 
 XERCES_CPP_NAMESPACE_BEGIN
 
+// Template instantiations
+template class XMLUTIL_EXPORT ArrayJanitor<XMLByte>;
+template class XMLUTIL_EXPORT ArrayJanitor<XMLCh>;
+
 // ---------------------------------------------------------------------------
 //  Local, static data
 //
--- ./src/xercesc/util/TransService.hpp.orig	2018-02-15 02:22:38.000000000 +0100
+++ ./src/xercesc/util/TransService.hpp	2018-06-09 00:48:16.000000000 +0200
@@ -461,9 +461,6 @@ private :
     XMLLCPTranscoder& operator=(const XMLLCPTranscoder&);
 };
 
-template class XMLUTIL_EXPORT ArrayJanitor<XMLByte>;
-template class XMLUTIL_EXPORT ArrayJanitor<XMLCh>;
-
 //
 // This class can be used to transcode to a target encoding. It manages the
 // memory allocated for the transcode in an exception safe manner, automatically
