--- src/clock.cpp.orig	2017-02-18 18:48:41.000000000 +0100
+++ src/clock.cpp	2017-04-23 04:23:24.000000000 +0200
@@ -241,7 +241,11 @@ uint64_t zmq::clock_t::rdtsc ()
 #else
     struct timespec ts;
     #if defined ZMQ_HAVE_OSX && __MAC_OS_X_VERSION_MIN_REQUIRED < 101200 // less than macOS 10.12
-        alt_clock_gettime (CLOCK_MONOTONIC, &ts);
+        #ifdef CLOCK_MONOTONIC
+            alt_clock_gettime (CLOCK_MONOTONIC, &ts);
+        #else
+            alt_clock_gettime (CLOCK_REALTIME, &ts);
+        #endif
     #else
         clock_gettime (CLOCK_MONOTONIC, &ts);
     #endif
