--- src/color-profile.cpp.orig	2017-01-02 10:44:59.000000000 +0100
+++ src/color-profile.cpp	2017-01-14 00:49:47.000000000 +0100
@@ -939,7 +939,12 @@ Glib::ustring getNameFromProfile(cmsHPRO
         std::vector<char> data(byteLen);
         cmsUInt32Number readLen = cmsGetProfileInfoASCII(profile, cmsInfoDescription,
                                                          "en", "US",
-                                                         data.data(), data.size());
+#if __cplusplus < 201103L
+                                                         &data[0],
+#else
+                                                         data.data(),
+#endif
+                                                         data.size());
         if (readLen < data.size()) {
             data.resize(readLen);
         }
