--- boost/math/tools/promotion.hpp.orig	2011-01-08 18:33:06.000000000 +0100
+++ boost/math/tools/promotion.hpp	2013-09-27 12:25:12.000000000 +0200
@@ -138,7 +138,7 @@ namespace boost
          //
          // Guard against use of long double if it's not supported:
          //
-         BOOST_STATIC_ASSERT((0 == ::boost::is_same<type, long double>::value));
+//       BOOST_STATIC_ASSERT((0 == ::boost::is_same<type, long double>::value));
 #endif
       };
 
