--- Rendering/OpenGL/vtkCocoaMacOSXSDKCompatibility.h.orig	2018-12-22 18:38:16.000000000 +0100
+++ Rendering/OpenGL/vtkCocoaMacOSXSDKCompatibility.h	2018-12-22 19:27:00.000000000 +0100
@@ -24,12 +24,12 @@ PURPOSE.  See the above copyright notice
 
 #include <AvailabilityMacros.h>
 
-#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
-  #error VTK requires the Mac OS X 10.7 SDK or later
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1040
+  #error VTK requires the Mac OS X 10.4 SDK or later
 #endif
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED < 1070
-  #error VTK requires a deployment target of Mac OS X 10.7 or later
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1040
+  #error VTK requires a deployment target of Mac OS X 10.4 or later
 #endif
 
 #if (MAC_OS_X_VERSION_MAX_ALLOWED < 101200) && !defined(VTK_DONT_MAP_10_12_ENUMS)
--- Rendering/OpenGL2/vtkCocoaMacOSXSDKCompatibility.h.orig	2018-12-22 18:38:16.000000000 +0100
+++ Rendering/OpenGL2/vtkCocoaMacOSXSDKCompatibility.h	2018-12-22 19:27:00.000000000 +0100
@@ -24,12 +24,12 @@ PURPOSE.  See the above copyright notice
 
 #include <AvailabilityMacros.h>
 
-#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
-  #error VTK requires the Mac OS X 10.7 SDK or later
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1040
+  #error VTK requires the Mac OS X 10.4 SDK or later
 #endif
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED < 1070
-  #error VTK requires a deployment target of Mac OS X 10.7 or later
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1040
+  #error VTK requires a deployment target of Mac OS X 10.4 or later
 #endif
 
 #if (MAC_OS_X_VERSION_MAX_ALLOWED < 101200) && !defined(VTK_DONT_MAP_10_12_ENUMS)
