--- dir.c.orig	2016-11-05 16:49:29.000000000 +0100
+++ dir.c	2017-03-01 12:32:50.000000000 +0100
@@ -1515,7 +1515,7 @@ join_path(const char *path, long len, in
     return buf;
 }
 
-#ifdef HAVE_GETATTRLIST
+#ifdef HAVE_FGETATTRLIST
 static int
 is_case_sensitive(DIR *dirp)
 {
@@ -1534,7 +1534,9 @@ is_case_sensitive(DIR *dirp)
 	return -1;
     return (cap->capabilities[idx] & mask) != 0;
 }
+#endif
 
+#ifdef HAVE_GETATTRLIST
 static char *
 replace_real_basename(char *path, long base, rb_encoding *enc, int norm_p, int flags, rb_pathtype_t *type)
 {
@@ -1823,7 +1825,7 @@ glob_helper(
 	    goto literally;
 	}
 # endif
-# ifdef HAVE_GETATTRLIST
+# ifdef HAVE_FGETATTRLIST
 	if (is_case_sensitive(dirp) == 0)
 	    flags |= FNM_CASEFOLD;
 # endif
