--- src/plugins/notification/Makefile.am.orig	2015-10-11 10:19:33.000000000 +0200
+++ src/plugins/notification/Makefile.am	2015-11-12 19:09:34.000000000 +0100
@@ -13,7 +13,7 @@
 
 SUBDIRS=gtkhotkey
 EXTRA_DIST = claws.def plugin.def version.rc
-CFLAGS += "-Wall"
+AM_CFLAGS = "-Wall"
 
 IFLAGS = \
 	-I$(top_srcdir)/src/plugins/notification/gtkhotkey \
--- src/plugins/vcalendar/libical/libical/Makefile.am.orig	2015-10-11 10:19:33.000000000 +0200
+++ src/plugins/vcalendar/libical/libical/Makefile.am	2015-11-12 19:07:23.000000000 +0100
@@ -30,8 +30,8 @@
 noinst_LTLIBRARIES = libical.la
 endif
 
-YFLAGS = -d -v -t -pical_yy
-LFLAGS = -Pical_yy
+AM_YFLAGS = -d -v -t -pical_yy
+AM_LFLAGS = -Pical_yy
 LEX_OUTPUT_ROOT = lex.ical_yy
 
 libical_la_LDFLAGS = -version-info 0:0:0
